service/history/shardController.go

```diff
@@ -106,15 +109,15 @@ func newShardController(host *membership.HostInfo, resolver membership.ServiceRe
 }
 
 func newHistoryShardsItem(shardID int, shardMgr persistence.ShardManager, historyMgr persistence.HistoryManager,
-	metadataMgr persistence.MetadataManager, executionMgrFactory persistence.ExecutionManagerFactory, factory EngineFactory,
-	host *membership.HostInfo, config *Config, logger bark.Logger, metricsClient metrics.Client) (*historyShardsItem, error) {
+	metadataMgr persistence.MetadataManager, executionMgrFactory persistence.ExecutionManagerFactory, clusterMetadata cluster.Metadata,
+	factory EngineFactory, host *membership.HostInfo, config *Config, logger bark.Logger, metricsClient metrics.Client) (*historyShardsItem, error) {
 
 	executionMgr, err := executionMgrFactory.CreateExecutionManager(shardID)
 	if err != nil {
 		return nil, err
 	}
 
-	domainCache := cache.NewDomainCache(metadataMgr, logger)
+	domainCache := cache.NewDomainCache(metadataMgr, clusterMetadata, logger)
```

Comment:
Why do we have a cache per shard?  I think domain cache should be created for the history host rather than instantiated per shard.

Reply:
the cache is implemented using just one mutex, as shown below, so if the lock is not a concern, i can do the update:
```
lru struct {
	mut      sync.Mutex
	byAccess *list.List
	byKey    map[interface{}]*list.Element
	maxSize  int
	ttl      time.Duration
	pin      bool
	rmFunc   RemovedFunc
}
```